src/jwt/mod.rs

```diff
@@ -20,11 +20,98 @@ new_type![
     JsonWebTokenContentType(String)
 ];
 
+#[derive(Error, Debug)]
+pub enum JsonWebTokenTypeNormalizeError {
+    #[error("Has parameters set without application specifies like JWT;param=true (should be application/JWT;param=true)")]
+    ParameterWithoutApplication(),
+
+    #[error(
+        "The web token type is complex (contains /) but does not start with an application type."
+    )]
+    ComplexTypeWithoutApplication(),
+}
+
 new_type![
+    /// JSON web token type field (typ)
+    ///
+    /// This types stores the raw (deserialized) value.
+    ///
+    /// To compare two different JSON Web Token types, please use the normalized version via [`JsonWebTokenType::normalize`]
     #[derive(Deserialize, Hash, Ord, PartialOrd, Serialize)]
     JsonWebTokenType(String)
+
+    impl {
+        /// Expands a [`JsonWebTokenType`] and produces a [`NormalizedJsonWebTokenType`] according to RFC2045 and RFC7515.
+        ///
+        /// See [RFC 2045 section 5.1](https://tools.ietf.org/html/rfc2045#section-5.1) for the full Content-Type Header Field spec.
+        /// See [RFC 7515 section 4.19](https://tools.ietf.org/html/rfc7515#section-4.1.9) for specific requirements of JSON Web Token Types.
+        pub fn normalize(&self) -> Result<NormalizedJsonWebTokenType, JsonWebTokenTypeNormalizeError> {
+            self.try_into()
+        }
+    }
+];
+
+new_type![
+    /// JSON web token type field (typ)
+    ///
+    /// This types stores the raw (deserialized) value.
+    /// To retrieve a normalized value according to RFC2045 and RFC7515 see [`JsonWebTokenType::normalize`]
+    ///
+    /// See [RFC 2045 section 5.1](https://tools.ietf.org/html/rfc2045#section-5.1) for the full Content-Type Header Field spec.
+    /// See [RFC 7515 section 4.1.9](https://tools.ietf.org/html/rfc7515#section-4.1.9) for specific requirements of JSON Web Token Types.
+    ///
+    /// It is recommended to only instantiate `NormalizedJsonWebTokenType` objects via [`JsonWebTokenType`] and then call [`JsonWebTokenType::normalize`].
+    ///
+    /// ```rust
+    /// # use openidconnect::{NormalizedJsonWebTokenType, JsonWebTokenType};
+    /// let token_type: NormalizedJsonWebTokenType = JsonWebTokenType::new("jwt+at".to_string())
+    ///     .normalize()
+    ///     .unwrap();
```

Comment:
I'd rather not include `.unwrap()` in the documentation snippet since it encourages users not to properly handle errors. can we wrap it in a (hidden using `#`) function that returns a `Result` and then change this to `.normalize()?`?

Reply:
I am actually not sure what you mean. I removed the `unwrap` from shown documentation at least.